src/core/exec-invoke.c

```diff
@@ -4174,7 +4174,7 @@ static void log_command_line(
                    LOG_EXEC_INVOCATION_ID(params));
 }
 
-static bool exec_context_needs_cap_sys_admin(const ExecContext *context) {
+static bool exec_context_needs_cap_sys_admin(const ExecContext *context, const ExecParameters *params) {
```

Comment:
maybe rename this func now, after all it's not just an operation on ExecContext anymore, but one that takes two structs into account.

maybe just call this exec_needs_cap_sys_admin(), dropping the "context"